src/lib.rs

```diff
@@ -1137,29 +1554,36 @@ impl<'a> ComputePass<'a> {
         indirect_buffer: &'a Buffer,
         indirect_offset: BufferAddress,
     ) {
-        backend::compute_pass_dispatch_indirect(&self.id, &indirect_buffer.id, indirect_offset);
+        self.id
+            .dispatch_indirect(&indirect_buffer.id, indirect_offset);
     }
 }
 
 impl<'a> Drop for ComputePass<'a> {
     fn drop(&mut self) {
         if !thread::panicking() {
-            backend::compute_pass_end_pass(&self.id);
+            self.parent
+                .context
+                .encoder_end_compute_pass(&self.parent.id, &mut self.id);
         }
     }
 }
 
 impl Queue {
     /// Submits a series of finished command buffers for execution.
-    pub fn submit(&self, command_buffers: &[CommandBuffer]) {
-        backend::queue_submit(&self.id, command_buffers);
+    pub fn submit<I: IntoIterator<Item = CommandBuffer>>(&self, command_buffers: I) {
```

Comment:
Couldn't this be inefficient for webgpu.h, since that takes a pointer to a buffer of command buffers? You'd have to allocate a Vec or similar to use that. Whereas, a slice could just be passed efficiently.

Reply:
Nah, this is more flexible than a slice. You can always go from a slice to an iterator for free, but not the other way around. I.e. `wgpu-native` would do something like: `command_buffer_slice.iter().cloned()` for passing it here.